tests/components/vicare/test_config_flow.py

```diff
@@ -86,22 +79,15 @@ async def test_form_dhcp(hass: HomeAssistant) -> None:
     assert result["step_id"] == "user"
     assert result["errors"] == {}
 
-    with patch(
-        "homeassistant.components.vicare.config_flow.vicare_login",
-        return_value=None,
-    ), patch(
-        "homeassistant.components.vicare.async_setup_entry",
-        return_value=True,
-    ) as mock_setup_entry:
-        result2 = await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            {
-                CONF_USERNAME: "foo@bar.com",
-                CONF_PASSWORD: "1234",
-                CONF_CLIENT_ID: "5678",
-            },
-        )
-        await hass.async_block_till_done()
+    result2 = await hass.config_entries.flow.async_configure(
```

Comment:
You should keep the `vicare_login` patch here.

Reply:
Hi @epenet and thanks for taking the time to review this!

I'm not sure I got this right:
I replaced vicare_login with fixtures that return MockPyViCare. This is the purpose of that test refactoring: MockPyViCare allows us to use API json dumps instead of the real API.
I would like to use these mocks in all tests to simulate different heating devices and check the behavior of config flow and later sensors entities, etc.

Why should I keep the vicare_login patch? If I don't introduce the mock there won't be much left in the preliminary PR for conflig flow test refactoring ;)

Reply:
If it's smaller it's easier to review.

Reply:
Here's the preliminary config_flow test:
https://github.com/home-assistant/core/pull/90568